dask/array/core.py

```diff
@@ -769,7 +771,7 @@ def cache(self, store=None, **kwargs):
                         "1. Install ``chest``, an out-of-core dictionary\n"
                         "2. Provide an on-disk array like an h5py.Dataset") # pragma: no cover
         if isinstance(store, MutableMapping):
-            name = next(names)
+            name = 'cache-' + tokenize((self.name, store))
```

Comment:
The store may be a numpy array.  Although I suppose that taking the `str` of a numpy array isn't that bad.